packages/playwright-core/src/server/fetch.ts

```diff
@@ -491,8 +498,14 @@ export abstract class APIRequestContext extends SdkObject {
       request.on('socket', socket => {
         // happy eyeballs don't emit lookup and connect events, so we use our custom ones
         const happyEyeBallsTimings = timingForSocket(socket);
-        dnsLookupAt = happyEyeBallsTimings.dnsLookupAt;
-        tcpConnectionAt ??= happyEyeBallsTimings.tcpConnectionAt;
+        if (request.reusedSocket) {
+          reusedSocketAt = monotonicTime();
+          dnsLookupAt = startAt;
```

Comment:
Perhaps set both `dnsLookupAt` and `tcpConnectionAt` to `undefined`? Would that make the code cleaner?

Reply:
Good idea. I've refactored it a little more, how do you like that?